src/main/java/duke/exception/DukeException.java

```diff
@@ -0,0 +1,25 @@
+package duke.exception;
+
+/**
+ * Encapsulates exceptions that are specific to Duke chat bot. Most of these exceptions are
+ * user's input error.
+ */
+public class DukeException extends Exception {
+    /**
+     * The prefix of the error message of all DukeException objects.
+     */
+    public static final String PREFIX =  "\u2639  OOPS!!!"; // The unicode is a sad face symbol.
+
+    public DukeException() {
+        super(String.format("%s DukeException occurs!", PREFIX));
+    }
+
```

Comment:
I would add the following Constructor here:

`    public DukeException(String message, Throwable cause) {
        super(message, cause);
    }
`

What this does is it allows you to catch the underlying exception in the `catch` block, then re-throw a DukeException together with the real cause. This is reflected in the stack trace which provides the developer with a better inkling as to what exactly caused the error in the `try` block.

Again, our app is trivial and may not require that much finesse but hey, it'll help when we do AddressBook!

Reply:
Alright, thanks a lot for the suggestion!